transition-scripts/helpers/keycloak.sh

```diff
@@ -32,3 +32,191 @@ wait_for_keycloak_all_ready() {
+}
+
+get_keycloak_admin_token() {
+  if [ "$#" -lt 2 ]; then exit 1; fi
+
+  cluster="$1"
+  namespace="$2"
+  ensure_kube_context "$cluster"
+
+  username=$(kubectl get secret sso-keycloak-admin -n "$namespace" -o jsonpath='{.data.username}' | base64 -d)
+  password=$(kubectl get secret sso-keycloak-admin -n "$namespace" -o jsonpath='{.data.password}' | base64 -d)
+
+  read -r status_code data < <(curl_http_code \
+    -d "client_id=admin-cli" \
+    -d "username=$username" \
+    -d "password=$password" \
+    -d "grant_type=password" \
+    "$(get_ocp_keycloak_url "$cluster" "$namespace")/auth/realms/master/protocol/openid-connect/token")
+
+  if [ "$status_code" -ne "200" ] || [ "$data" == null ]; then
+    warn "failed to fetch Keycloak admin access token"
+    exit 1
+  fi
+
+  access_token=$(echo "$data" | jq -r '.access_token')
+
+  echo "$access_token"
+}
+
+curl_keycloak_api() {
+  if [ "$#" -lt 2 ]; then exit 1; fi
+
+  cluster="$1"
+  namespace="$2"
+  ensure_kube_context "$cluster"
+
+  access_token=$(get_keycloak_admin_token "$cluster" "$namespace")
+
+  read -r status_code data < <(curl_http_code \
+    -H "Authorization: Bearer $access_token" \
+    -H "Content-Type: application/json" "${@:3}")
+
+  echo "$status_code" "$data"
+}
+
+create_keycloak_realm() {
+  if [ "$#" -lt 3 ]; then exit 1; fi
+
+  cluster="$1"
+  namespace="$2"
+  realmname="$3"
+  ensure_kube_context "$cluster"
+
+  read -r status_code data < <(curl_keycloak_api "$cluster" "$namespace" -X POST \
+    -d '{"realm":'\""$realmname\""'}' \
+    "$(get_ocp_keycloak_url "$cluster" "$namespace")/auth/admin/realms")
+
+  if [ "$status_code" -ne "201" ]; then
+    error "$status_code: failed to create a new realm"
+    error "$data"
+    exit 1
+  fi
+
+  echo "success"
+}
+
+get_keycloak_realm() {
+  if [ "$#" -lt 3 ]; then exit 1; fi
+
+  cluster="$1"
+  namespace="$2"
+  realmname="$3"
+  ensure_kube_context "$cluster"
+
+  read -r status_code data < <(curl_keycloak_api "$cluster" "$namespace" -X GET \
+    "$(get_ocp_keycloak_url "$cluster" "$namespace")/auth/admin/realms/$realmname")
+
+  if [ "$status_code" -ne "200" ]; then
+    return
+  fi
+
+  echo "$data"
+}
+
+list_keycloak_realms() {
+  if [ "$#" -lt 2 ]; then exit 1; fi
+
+  cluster="$1"
+  namespace="$2"
+  ensure_kube_context "$cluster"
+
+  read -r status_code data < <(curl_keycloak_api "$cluster" "$namespace" -X GET \
+    "$(get_ocp_keycloak_url "$cluster" "$namespace")/auth/admin/realms")
+
+  if [ "$status_code" -ne "200" ]; then
+    return
+  fi
+
+  echo "$data"
+}
+
+delete_keycloak_realm() {
+  if [ "$#" -lt 3 ]; then exit 1; fi
+
+  cluster="$1"
+  namespace="$2"
+  realmname="$3"
+  ensure_kube_context "$cluster"
+
+  read -r status_code data < <(curl_keycloak_api "$cluster" "$namespace" -X DELETE \
+    "$(get_ocp_keycloak_url "$cluster" "$namespace")/auth/admin/realms/$realmname")
+
+  if [ "$status_code" -ne "204" ]; then
+    return
+  fi
+
+  echo "$data"
+}
+
+remove_all_realms() {
```

Comment:
Does this mean each time the tests are run the sandbox dev deployment will be nuked of realms?

Reply:
yes, as each test would create 10 realms, it needs to remove all test realms before it runs.